test/parallel/test-domain-intercept.js

```diff
@@ -0,0 +1,43 @@
+'use strict';
+
+const common = require('../common');
+const assert = require('assert');
+const domain = require('domain');
+
+{
+  const d = new domain.Domain();
+
+  const mustNotCall = common.mustNotCall();
+
+  d.on('error', common.mustCall((err) => {
+    assert.strictEqual(err.message, 'foobar');
+    assert.strictEqual(err.domain, d);
+    assert.strictEqual(err.domainEmitter, undefined);
+    assert.strictEqual(err.domainBound, mustNotCall);
+    assert.strictEqual(err.domainThrown, false);
+  }));
+
+  const bound = d.intercept(mustNotCall);
+  bound(new Error('foobar'));
+}
+
+{
+  const d = new domain.Domain();
+
+  const bound = d.intercept(common.mustCall((data) => {
```

Comment:
The problem I see with the previous comments regarding this test (and some of the others) is that they seemed more like API documentation rather than actually helping to explain what the test does. In this case, I hope it becomes much more visible what each part of the test does by identifying the groups of code that test a single piece of the functionality.